apps/desktop/src/lib/shared/LabelInput.svelte

```diff
@@ -0,0 +1,172 @@
+<script lang="ts">
+	import { autoSelectBranchNameFeature } from '$lib/config/uiFeatureFlags';
+	import { debounce } from '$lib/utils/debounce';
+	import { resizeObserver } from '$lib/utils/resizeObserver';
+
+	interface Props {
+		value: string;
+		disabled?: boolean;
+		onChange?: (value: string) => void;
+		placeholder?: string;
+		class?: string;
+		multiline?: boolean;
+		inputEl?: HTMLInputElement | HTMLTextAreaElement;
+	}
+
+	let {
+		value,
+		placeholder,
+		disabled = false,
+		onChange,
+		inputEl = $bindable(),
+		class: className = '',
+		multiline = false
+	}: Props = $props();
+
+	let initialValue = value;
+	let textAreaWidth = $state('');
+	let inputWidth = $state('');
+	let inputHeight = $state('');
+</script>
+
+<span
+	use:resizeObserver={debounce((e) => {
+		inputWidth = `${Math.round(e.frame.width)}px`;
+		inputHeight = `${Math.round(e.frame.height)}px`;
+	}, 100)}
+	class="label-input-measure-el"
+	class:text-12={multiline}
+	class:text-14={!multiline}
+	class:wrap={multiline}
+	style:width={multiline ? textAreaWidth : 'inherit'}
+>
+	{value}
+</span>
+{#if multiline}
```

Comment:
The previous `svelte:component` dynamic component mechanism is deprecated and the new dynamic component stuff didnt work either, i.e. something like

```tsx
InputComponent = multiline ? new HTMLTextAreaElement() : new HTMLInputElement()
``` 

and neither did the `document.createElement()` variant :( 

_I think_ it only works with Svelte components, not vanilla html elements. But I'd love to simplify this some more, the `textarea`/`input` are so similar it feels like a waste having a giant `if` here, so I'm open to any ideas

Reply:
`is` would be wonderful, but safari doesn't support it D: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/is

Reply:
You could define a svelte component with is `TextInput`, which you can pass all the arguments to. It can then be like:

```svelte
<script lang="ts">
  interface Props { ... }

  const { multiline, ...args }: Props = $props()
</script>

{#if multiline}
  <textarea {...args}></textarea>
{:else}
  <input {...args}></input>
{/if}
```

Reply:
OOrrrrrrr, create a svelte component wrapper for `<Input />` and `<TextArea />` and then use svelte's "correct" mechanism on those.

Reply:
Yeah was thinking about a wrapper component too, but then I figured we're just adding another layer of abstraction on top of the same `if` block, but I overlooked the ability to spread all the props in now. I like this idea!